File: algorithms/448. Find All Numbers Disappeared in an Array/solution.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i, n in enumerate(nums):
            while n:
                nums[n-1], n = 0, nums[n-1]
        return [i+1 for i, n in enumerate(nums) if n]

File: algorithms/448. Find All Numbers Disappeared in an Array/solution.cc
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for ( int i = 0; i < nums.size(); i++ ) {
            int next = abs(nums[i]) - 1;
            if ( nums[next] > 0 ) {
                nums[next] = -nums[next];
            }
        }
        vector<int> result;
        for ( int i = 0; i < nums.size(); i++ ) {
            if ( nums[i] > 0 ) {
                result.push_back(i + 1);
            }
        }
        return result;
    }
};
